include/FreeRTOS.h

```diff
@@ -1268,6 +1268,10 @@
     #define configRUN_ADDITIONAL_TESTS    0
 #endif
 
+#ifndef configIDLE_TASK_HOOK
+    #define configIDLE_TASK_HOOK()
```

Comment:
I do have a couple of concerns here.
The functions vApplicationIdleHook() and vApplicationMinimalIdleHook() already exist, so I think this would be rather confusing to users as to why this option now exists.

Using configIDLE_TASK_HOOK() as a means of inserting a 'break' into the loop also doesn't seem like a very clean approach. It also means the CMOK stub/override function effectively has to have the same behavior for each test file.

IMO, I would suggest something like changing the for() to a do .. while() where you can do something like: 

#define INFINITE_LOOP()   1
while(INFINITE_LOOP() )

For unit testing you can redefine the macro as:
#define INFINITE_LOOP()    mockInfLoop()
Which can now have a default behavior (returns 1) or be implemented differently on a per test basis (has a loop counter or immediately returns 0). This is also a more generic solution that can be used anywhere else in the kernel where an infinite loop exists.